src/services/staking_vaults.py

```diff
@@ -455,6 +478,17 @@ def calc_liquidity_fee(
                     minted_steth_on_event, blocks_elapsed_between_events, core_apr_ratio, liquidity_fee
                 )
 
+                if isinstance(event, VaultConnectedEvent):
+                    # If we catch a VaultConnectedEvent, it means that in the past there could be no more events,
+                    # because the vault was previously disconnected.
+                    # Technically, we could skip this check, but it explicitly communicates the business logic and intention.
+                    if vault_liquidity_fee != 0:
+                        raise ValueError(
+                            f"Wrong vault liquidity fee by vault {vault_address}. Vault had reconnected event and then his vault_liquidity_fee must be 0. got {vault_liquidity_fee}"
+                        )
+
+                    return vault_liquidity_fee, liability_shares
```

Comment:
The `vault_liquidity_fee` can be > 0 here, because it's calculated backwards.
The `liabilityShares` must be 0.

Reply:
fixed here: https://github.com/lidofinance/lido-oracle/pull/734/commits/a0dd17dd0f58556d003ff6a477eff79900a5fad9